### PR TITLE
feat(auth-api-lib): Add support for api scope groups

### PR DESCRIPTION
**Pull request by unnarb**
#ApiScopeGroup, library.. models and service

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/auth-api-lib/migrations/20210519134445-api-scope-groups.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          id UUID NOT NULL,
+          name VARCHAR NOT NULL,
+          description VARCHAR NOT NULL,
+          created TIMESTAMP WITH TIME ZONE DEFAULT now(),
+          modified TIMESTAMP WITH TIME ZONE,
+          PRIMARY KEY (id)
+        );
+
+        ALTER TABLE api_scope 
+        ADD COLUMN api_scope_group UUID NULL;
```
Comment (valurefugl) [inline libs/auth-api-lib/migrations/20210519134445-api-scope-groups.js:19]: Maybe 'group' would be a more convenient name for the column in the 'api_scope' table?

Comment (valurefugl) [reply libs/auth-api-lib/migrations/20210519134445-api-scope-groups.js:19]: or group_id ?

Comment (unnarb) [reply libs/auth-api-lib/migrations/20210519134445-api-scope-groups.js:19]: Sounds good

File: libs/auth-api-lib/src/lib/entities/dto/api-scopes.dto.ts [hunk 2]
```diff
@@ -29,6 +29,13 @@ export class ApiScopesDTO {
   })
   readonly description!: string
 
+  @IsOptional()
+  @IsString()
+  @ApiProperty({
+    example: '123e4567-e89b-12d3-a456-426614174000',
+  })
+  readonly groupId!: string
```
Comment (valurefugl) [inline libs/auth-api-lib/src/lib/entities/dto/api-scopes.dto.ts:37]: Should probably be nullable?

Comment (unnarb) [reply libs/auth-api-lib/src/lib/entities/dto/api-scopes.dto.ts:37]: of course

**Review by valurefugl — APPROVED** [review]

**Pull request merged**